tools/commands-packages-query.js

```diff
@@ -0,0 +1,1159 @@
+        return v.publishedOn;
+      });
+      Console.info("Experimental versions:");
+      var rows = [];
+      _.each(experimentalVersions, function (vr) {
+        // Experimental versions cannot be recommended.
+        rows.push([vr.version, utils.longformDate(vr.publishedOn)]);
+      });
+      Console.printTwoColumns(rows, columnOpts);
+    }
+  }
+});
+
+main.registerCommand({
+  name: 'show',
+  pretty: true,
+  minArgs: 1,
+  maxArgs: 1,
+  usesPackage: true,
+  options: {
+    "show-all": { type: Boolean, required: false },
+    "ejson": { type: Boolean, required: false }
+  },
+  catalogRefresh:
+    new catalog.Refresh.OnceAtStart(
+        { maxAge: DEFAULT_MAX_AGE_MS, ignoreErrors: true })
+}, function (options) {
+  // The foo@bar API means that we have to do some string parsing to figure out
+  // if we want a particular version.
+  var itemName = options.args[0];
+  var splitArgs = itemName.split('@');
+  var name = splitArgs[0];
+  var version = (splitArgs.length > 1) ? splitArgs[1] : null;
+  var query = null;
+
+  // Because of the new projectContext interface, we need to initialize the
+  // project context in order to load the local catalog. This is not ideal.
+  var projectContext = getTempContext(options);
+
+  // First, we need to figure out if we are dealing with a package, or a
+  // release. We don't want to rely on capitalization conventions, so we will
+  // start by checking if a package by that name exists. If it does, then we
+  // are dealing with a package.
+  var packageRecord =
+        catalog.official.getPackage(name) ||
+        projectContext.localCatalog.getPackage(name);
+  if (packageRecord) {
+    query =  new PackageQuery({
+      record: packageRecord,
+      version: version,
+      projectContext: projectContext,
+      showHiddenVersions: options["show-all"],
+      showOSArchitectures: options.ejson,
+      showDependencies: !! version
+    });
+  }
+
+  // If this is not a package, it might be a release. Let's check if there is
+  // a release by this name. There are no local releases, so we only need to
+  // check the official catalog.
+  if (! query) {
+    var releaseRecord = catalog.official.getReleaseTrack(name);
+    if (releaseRecord) {
+      query = new ReleaseQuery({
+        record: releaseRecord,
+        version: version,
+        showHiddenVersions: options["show-all"]
+      });
+    }
+  }
+  // If we have failed to create a query, or if we have created a query and it
+  // couldn't gather any data about our request, then the item that we are
+  // looking for does not exist.
+  if (! query || ! query.data) {
+    return itemNotFound(itemName);
+  }
+
+  query.print({ ejson: !! options.ejson });
+  return 0;
+});
+
+
+///////////////////////////////////////////////////////////////////////////////
+// search
+///////////////////////////////////////////////////////////////////////////////
+
+main.registerCommand({
+  name: 'search',
+  pretty: true,
```

Comment:
Probably should use the just-merged-to-devel "pretty can be a function" support to turn off pretty when ejson is set (for both show and search)